Makefile

```diff
@@ -11,57 +11,78 @@ else
+.PHONY: build build-darwin build-linux build-all clean fmt vet test acceptance-e2e acceptance-run acceptance-prepare acceptance-destory generate manifests install-tools deploy
 
 build: $(PROG)
-build-darwin: $(PROG:=.darwin_amd64)
-build-linux: $(PROG:=.linux_amd64)
+build-darwin: $(PROG:=.darwin)
+build-linux: $(PROG:=.linux)
 build-all: build-darwin build-linux
 
 # Specific linux build target, making it easy to work with the docker acceptance
-# tests on OSX
-bin/%.linux_amd64:
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(BUILD_COMMAND) -a -o $@ ./cmd/$*/.
+# tests on OSX. It uses by default your local arch
+bin/%.linux:
+	CGO_ENABLED=0 GOOS=linux $(BUILD_COMMAND) -a -o $@ ./cmd/$*/.
 
-bin/%.darwin_amd64:
-	CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(BUILD_COMMAND) -a -o $@ ./cmd/$*/.
+bin/%.darwin:
+	CGO_ENABLED=0 GOOS=darwin $(BUILD_COMMAND) -a -o $@ ./cmd/$*/.
 
 bin/%:
-	CGO_ENABLED=0 GOARCH=amd64 $(BUILD_COMMAND) -o $@ ./cmd/$*/.
-
-# Run the below commands in order to install the required dependencies for
-# running `make test` locally.
-# go install github.com/onsi/ginkgo/ginkgo@v1.16.5
-# go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
-# setup-envtest use -p path 1.22.x
-# source <(setup-envtest use -i -p env 1.22.x)
-test:
-	ginkgo -race -r ./...
+	CGO_ENABLED=0 $(BUILD_COMMAND) -o $@ ./cmd/$*/.
+
+clean:
+	rm -rvf $(PROG) $(PROG:%=%.linux) $(PROG:%=%.darwin)
+
+fmt:
+	go fmt ./...
 
 vet:
-	go vet ./cmd/rbac-manager/...
-	go vet ./cmd/theatre-consoles/...
-	go vet ./cmd/theatre-secrets/...
-	go vet ./cmd/vault-manager/...
-	go vet ./cmd/workloads-manager/...
+	go vet -tests -unreachable ./...
+
+test:
+	KUBEBUILDER_ASSETS="$(shell setup-envtest use -i -p path 1.24.x)" ginkgo -race -r ./...
+
+# Requires the following binaries: kubectl, kustomize, kind, docker
+acceptance-e2e: install-tools acceptance-prepare acceptance-run acceptance-destroy
+
+acceptance-run: install-tools
+	go run cmd/acceptance/main.go run --verbose
+
+acceptance-prepare: install-tools
+	go run cmd/acceptance/main.go prepare --verbose
 
-generate: controller-gen
-	$(CONTROLLER_GEN) object paths="./apis/rbac/..."
-	$(CONTROLLER_GEN) object paths="./apis/workloads/..."
+acceptance-destroy: install-tools
+	go run cmd/acceptance/main.go prepare --verbose
+
+generate: install-tools
+	controller-gen object paths="./apis/rbac/..."
+	controller-gen object paths="./apis/workloads/..."
 
 manifests: generate
 	$(CONTROLLER_GEN) crd paths="./apis/rbac/..." output:crd:artifacts:config=config/base/crds
 	$(CONTROLLER_GEN) crd paths="./apis/workloads/..." output:crd:artifacts:config=config/base/crds
 
+# See https://github.com/kubernetes-sigs/controller-runtime/tree/main/tools/setup-envtest
+install-tools:
+	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
+	go install github.com/onsi/ginkgo/ginkgo@v1.16.5
+	go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
```

Comment:
Can we make this a dependency for any target that requires these tools?

Reply:
Nvm, this is already the case. I only mentioned this because I failed to find gingko locally

Reply:
I think i missed the make test, fixing